playground/backend/internal/code_processing/code_processing.go

```diff
@@ -268,56 +334,75 @@ func finishByTimeout(ctx context.Context, pipelineId uuid.UUID, cacheService cac
 	cacheService.SetValue(ctx, pipelineId, cache.Status, pb.Status_STATUS_RUN_TIMEOUT)
 }
 
-// processError processes error received during processing code via setting a corresponding status and output to cache
-func processError(ctx context.Context, err error, data []byte, pipelineId uuid.UUID, cacheService cache.Cache, status pb.Status) {
-	switch status {
-	case pb.Status_STATUS_VALIDATION_ERROR:
-		logger.Errorf("%s: Validate: %s\n", pipelineId, err.Error())
+// processValidateError processes error received during processing validate step via setting a corresponding status and output to cache
+func processValidateError(ctx context.Context, errorChannel chan error, pipelineId uuid.UUID, cacheService cache.Cache) {
+	err := <-errorChannel
+	logger.Errorf("%s: Validate: %s\n", pipelineId, err.Error())
+
+	cacheService.SetValue(ctx, pipelineId, cache.Status, pb.Status_STATUS_VALIDATION_ERROR)
+}
+
+// processPrepareError processes error received during processing prepare step via setting a corresponding status and output to cache
+func processPrepareError(ctx context.Context, errorChannel chan error, pipelineId uuid.UUID, cacheService cache.Cache) {
+	err := <-errorChannel
+	logger.Errorf("%s: Prepare: %s\n", pipelineId, err.Error())
+
+	cacheService.SetValue(ctx, pipelineId, cache.Status, pb.Status_STATUS_PREPARATION_ERROR)
+}
```

Comment:
processValidateError and processPrepareError are almost identical, you can pass pb.Status and error message

Reply:
Changed.

Reply:
Probably, we should also add this to the docstring:
```
// processError processes error received during processing validation or preparation steps.
// This method sets corresponding status to the cache.
```